### PR TITLE
[IMP] product_packaging_dimension: Don't use max_weight anymore

### PR DESCRIPTION
**Pull request by rousseldenis**
As max_weight is defined now on stock.package.type, rely on that one.

Migrate data that was filled in to weight field.

Added string label to packaging_length too.

### CODE REVIEW

**Comment by rousseldenis** [comment]
@marielejeune @jbaudoux

File: product_packaging_dimension/models/product_packaging.py [hunk 1]
```diff
@@ -13,11 +13,6 @@ class ProductPackaging(models.Model):
         ("positive_height", "CHECK(height>=0)", "Height must be positive"),
         ("positive_width", "CHECK(width>=0)", "Width must be positive"),
         ("positive_length", "CHECK(packaging_length>=0)", "Length must be positive"),
-        (
-            "positive_max_weight",
-            "CHECK(max_weight>=0)",
-            "Max Weight must be positive",
```
Comment (jbaudoux) [inline product_packaging_dimension/models/product_packaging.py:19]: You could keep it but on weight instead of max_weight

Comment (rousseldenis) [reply product_packaging_dimension/models/product_packaging.py:19]: Done

**Review by jbaudoux — APPROVED** [review]

File: product_packaging_dimension/migrations/16.0.1.1.0/pre-migrate.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+def _move_max_weight(env):
+    """
+    Move max_weight value to weight if not defined
+    """
+    query = """
+        UPDATE product_packaging
+            SET weight = max_weight
+            WHERE weight IS NULL OR weight = 0.0
+            AND max_weight IS NOT NULL
```
Comment (jbaudoux) [inline product_packaging_dimension/migrations/16.0.1.1.0/pre-migrate.py]: Maybe only for the product packaging linked to a product.
```suggestion
            AND max_weight IS NOT NULL
            AND product_id IS NOT NULL
```

Comment (rousseldenis) [reply product_packaging_dimension/migrations/16.0.1.1.0/pre-migrate.py]: Done

**Comment by rousseldenis** [comment]
@marielejeune

**Comment by jbaudoux** [comment]
> As max_weight is defined now on stock.package.type, rely on that one.

Don't rely 😉

This PR relates to my comment here
https://github.com/OCA/product-attribute/pull/1161#issuecomment-1490120668

Cc @sebalix

**Review by lmignon — APPROVED** [review]

**Pull request merged**